test.py

```diff
@@ -787,8 +787,8 @@ def set_summary(summary):
                 # 1) failed pre-check, e.g. start failure
                 # 2) failed test execution.
                 if self.is_executed_ok is False:
-                    self.server_log = cluster.read_server_log()
-                    self.server_log_filename = cluster.server_log_filename()
+                    self.server_log = next(cluster.read_server_log())
```

Comment:
self.server_log and self.server_log_filename is type-hinted in several places, still `Optional[str]` / `Optional[pathlib.Path]`

Reply:
Also I'm worried that iterator may lead to bugs, what if it's used once, and then some other code wants to use it a second time? It's a trap. (The second time won't work)

Maybe list would be more appropriate?